### DISCUSSION TITLE
能不能实现一个这样的功能

### DISCUSSION THREAD

**Discussion by gongjianjvn** [alibaba/alibaba-rsocket-broker#185] [Q&A]
有三个区域, 内网, 前置机, 阿里云; 前置机可以访问内外网, 内网不能访问前置机. 现有的broker机制能不能将broker接收的特定group请求转发到 前置机的 client上, 前置机client再将请求转发到阿里云上的broker;

**Comment by pc859107393** [comment]
这个功能实际上是可以自行开发实现的，rsocket本身是双工通信，我们目前的一般模式都是从客户端访问服务端，但是别忘了咱这里面还有个广播功能，可以使用广播进行消息推送过去。

**Comment by pc859107393** [comment]
不过，个人建议有这种强需求的时候，可以考虑用消息队列之类的其他组件，避免大量的服务改造。

**Comment by gongjianjvn** [comment]
我在本地试了下, 通过自定义client端的RSocketResponderHandler, broker端的upstreamClusterRSocket基本可以实现. client端注册一个不存在的service, broker端匹配不到服务会调用upstreamClusterRSocket, 在这里用这个固定的service来定位下游注册的client; 然后转发请求, 下游的client收到后在本地的RSocketResponderHandler识别本地能不能处理, 不能再通过group转发到别的UpstreamClusterRSocket上;

**Reply by linux-china** [reply]
@gongjianjvn  目前的Broker cluster支持二级树状结构，就是一个总的parent cluster，然后下面是多个sub cluster，当sub cluster不能处理，就是转给parent cluster，然后parent cluster中在基于路由规则转发给sub cluster。
